frappe/public/js/frappe/form/controls/attach.js

```diff
@@ -77,8 +77,8 @@ frappe.ui.form.ControlAttach = class ControlAttach extends frappe.ui.form.Contro
 		};
 
 		if (this.frm) {
-			options.doctype = this.frm.doctype;
-			options.docname = this.frm.docname;
+			options.doctype = (this.frm.doctype==this.doctype) ? this.frm.doctype : this.doctype;
```

Comment:
This is not a valid fix. This will always set it on child doctype which will break other things - permission. 

A correct fix will require modifying the File doctype to specify complete fieldname reference: 
1. Doctype
2. document name
3. fieldname - can be main field or actual field
4. child table (fieldname of child table)
5. child doc name (row.name)
6.